### PR TITLE
RTP implementation

### PR DESCRIPTION
**Pull request by moninom1**
Issue #, if available:

RTP implementation.

Description of changes:

Add STUN serializer/deserializer and make them standalone. The change also adds codec packetizer and depacketizer for H.264, G.711, Opus and VP8 codec.

Testing -

RTP tests run with the wrapper layer for serialization and deserialization.
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice

### CODE REVIEW

**Review by thomas-roos — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,15 +1,75 @@
...
+as defined in [RFC 3550](https://datatracker.ietf.org/doc/html/rfc3550), is a data transport protocol
+used to deliver real-time data such as audio or video over IP networks. An RTP packet comprises of a
+header followed by the payload.
 
-## Security
+A RTP packet consist of following fields followed by an optional RTP Header extension.
 
-See [CONTRIBUTING](CONTRIBUTING.md#security-issue-notifications) for more information.
+```
+        0                   1                   2                   3
```
Comment (disa6302) [inline README.md:18]: Can we point to the RFC we referred to for future reference?

File: codec_packetizers/g711/g711_depacketizer.c [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        ( pFrame == NULL ) )
+    {
+        result = G711_RESULT_BAD_PARAM;
+    }
+
+    for( i = 0; ( i < pCtx->packetCount ) && ( result == G711_RESULT_OK ); i++ )
+    {
+        pPacket = &( pCtx->pPacketsArray[ i ] );
+
+        if( ( pFrame->frameDataLength - currentFrameDataIndex ) >= pPacket->packetDataLength )
```
Comment (disa6302) [inline codec_packetizers/g711/g711_depacketizer.c:83]: Any chance frameDataLength can we zero?

File: codec_packetizers/h264/h264_depacketizer.c [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+
+/*-----------------------------------------------------------*/
+
+H264Result_t H264Depacketizer_GetFrame( H264DepacketizerContext_t * pCtx,
+                                        Frame_t * pFrame )
+{
+    H264Result_t result = H264_RESULT_OK;
+    Nalu_t nalu;
+    size_t currentFrameDataIndex = 0;
+    uint8_t startCode[] = { 0x00, 0x00, 0x00, 0x01 };
```
Comment (disa6302) [inline codec_packetizers/h264/h264_depacketizer.c:209]: IIRC, the annex-b nalu can either have 0x00000001 or 0x000001. Is there a reason we are restricting the startCode to one of possibilities only?

Comment (moninom1) [reply codec_packetizers/h264/h264_depacketizer.c:209]: @disa6302 In create paylaod path we are using both [3byte](https://github.com/moninom1/amazon-kinesis-video-streams-rtp/blob/main/codec_packetizers/h264/h264_packetizer.c#L156C13-L156C23) and [4byte](https://github.com/moninom1/amazon-kinesis-video-streams-rtp/blob/main/codec_packetizers/h264/h264_packetizer.c#L155) code.
This is the depay path and here we are appending only 4byte code which is also inline with webrtc SDK [depayH264FromRtpPayload](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/master/src/source/Rtp/Codecs/RtpH264Payloader.c#L238) Let me know if it seems fine.

File: README.md (hunk 1 above)
Comment (moninom1) [reply README.md:18]: The RFC [RFC 3550](https://datatracker.ietf.org/doc/html/rfc3550), is mentioned just above this line. Is that fine? Or you are referring to different RFC?

File: codec_packetizers/g711/g711_depacketizer.c (hunk 2 above)
Comment (moninom1) [reply codec_packetizers/g711/g711_depacketizer.c:83]: Added zero length check.

File: README.md (hunk 1 above)
Comment (disa6302) [reply README.md:18]: Ah. Ok. I missed it then.

File: codec_packetizers/h264/h264_depacketizer.c (hunk 3 above)
Comment (disa6302) [reply codec_packetizers/h264/h264_depacketizer.c:209]: Got it. Ok. Wanted to confirm. Did not check the SDK implementation for this. Looks good!

**Review by disa6302 — APPROVED** [review]

**Review by ActoryOu — APPROVED** [review]

**Pull request merged**